src/composables/useTransactions.ts

```diff
@@ -396,25 +396,27 @@ export default function useTransactions() {
   }
 
   function checkTxActivity(transaction: Transaction) {
-    provider.value
-      .getTransactionReceipt(transaction.id)
-      .then(tx => {
-        if (tx != null) {
-          finalizeTransaction(transaction.id, 'tx', tx);
-        }
-      })
-      .catch(e =>
-        console.log(
-          '[Transactions]: Failed to fetch tx information',
-          transaction,
-          e
-        )
-      )
-      .finally(() =>
-        updateTransaction(transaction.id, 'tx', {
-          lastCheckedBlockNumber: blockNumber.value
+    if (provider.value != null) {
```

Comment:
I've added this guard here as well (which was causing the error in the first place)

Reply:
I think this might still fail because the exception is complaining that `provider` is undefined. Maybe it could be:
```suggestion
    if (provider?.value) {
```
or
```suggestion
    if (provider !== undefined) {
```